convert/HiggsDNA_to_FggFF.py

```diff
@@ -0,0 +1,218 @@
+    type = str,
+    default = "mva_score"
+)
+parser.add_argument(
+    "--mvas",
+    nargs="*",
+    help = "mva limits to SRs",
+    type = float,
+    default = [0.99,0.9]
+)
+parser.add_argument(
+    "--unblind",
+    help = "Unblind the signal region, 120-130 GeV",
+    action = "store_true"
+)
+args = parser.parse_args()
+
+"""
+Script to convert parquet files into FlashggFinalFit flavor TTrees.
+"""
+
+def to_tensor(dataframe, columns = [], dtypes = {}):
+    # Use all columns from data frame if none where listed when called
+    if len(columns) <= 0:
+        columns = dataframe.fields
+    # Build list of dtypes to use, updating from any `dtypes` passed when called
+    dtype_list = []
+    for column in columns:
+        if column not in dtypes.keys():
+            if column == "proc" or column == "sample"or column =="run_era":
+                dtype_list.append("S15")
+            else:
+                dtype_list.append(type(dataframe[column][0]))
+        else:
+            dtype_list.append(dtypes[column])
+    # Build dictionary with lists of column names and formatting in the same order
+    dtype_dict = {
+        'names': columns,
+        'formats': dtype_list
+    }
+    # Initialize _mostly_ empty numpy array with column names and formatting
+    numpy_buffer = numpy.zeros(
+        shape = len(dataframe),
+        dtype = dtype_dict
+    )
+    # Insert values from dataframe columns into numpy labels
+    for col in columns:
+        #print(f"column {col} input shape {numpy_buffer[col].shape} output shape {dataframe[col].to_numpy().shape}")
+        numpy_buffer[col] = dataframe[col].to_numpy()
+    return numpy_buffer
+
+def save_to_root(filePath, TTreeName, df):
+    if os.path.isfile(filePath):
+        with uproot.update(filePath) as f_out:
+            f_out[TTreeName] = df_sr
+    else:
+        with uproot.recreate(filePath) as f_out:
+            f_out[TTreeName] = df_sr
+
+# Format output directories
+out_dir = str(args.FggFF) + 'files_systs/' + str(args.tag) + '/'
+# Purge old files at save destination, set output structure
+os.system("rm -rf %s"%(out_dir))
+
+os.system("mkdir -p %s"%(out_dir))
+os.system("mkdir -p %s/Data"%(out_dir))
+os.system("mkdir -p %s/2016"%(out_dir))
+os.system("mkdir -p %s/2017"%(out_dir))
+os.system("mkdir -p %s/2018"%(out_dir))
+
+# Organize binning
+nSRs = len(args.mvas)
+args.mvas+=[99]
+args.mvas.sort(reverse=True)
+
+needed_fields=['CMS_hgg_mass','weight_central','process_id','proc']
+years = ['2017','2018']
+procs = {'ttHH_ggbb':'ttHH_ggbb',
+         'ttH_M125':'ttH'
+        }
+
+print('Started Processing Data')
+df = ak.from_parquet(str(args.input)+'merged_nominal.parquet')
+df = df[df.process_id == procs_id_map["Data"]]
+print(f'File contains {len(df)} total Data events')
+df['CMS_hgg_mass'] = df.Diphoton_mass
+
+out_file = f'{out_dir}/Data/allData.root'
+for sr in range(nSRs):
+    sr_mask = (df[args.mva_name] < args.mvas[sr]) & (df[args.mva_name] >= args.mvas[sr+1])
```

Comment:
Relevant especially if the comment about the `mvas` argument requires a change.
```suggestion
    sr_mask = (df[args.mva_name] <= args.mvas[sr]) & (df[args.mva_name] > args.mvas[sr+1])
```

Reply:
This is fine, but could depend on if the definition of the cut, eg inclusive vs exclusive